app/services/prometheus/service.go

```diff
@@ -17,39 +17,58 @@
 package prometheus
 
 import (
+	"errors"
+	"github.com/limechain/hedera-eth-bridge-validator/app/helper/metrics"
 	"github.com/limechain/hedera-eth-bridge-validator/config"
+	"github.com/limechain/hedera-eth-bridge-validator/constants"
 	"github.com/prometheus/client_golang/prometheus"
 	log "github.com/sirupsen/logrus"
+	"strconv"
+	"sync"
 )
 
 type Service struct {
-	logger *log.Entry
-	gauges map[string]prometheus.Gauge
+	mu                  sync.RWMutex
+	logger              *log.Entry
+	gauges              map[string]prometheus.Gauge
+	counters            map[string]prometheus.Counter
+	isMonitoringEnabled bool
+	assetsConfig        config.Assets
 }
 
-func NewService() *Service {
+func NewService(assetsConfig config.Assets, isMonitoringEnabled bool) *Service {
 
 	return &Service{
-		logger: config.GetLoggerFor("Prometheus Service"),
-		gauges: map[string]prometheus.Gauge{},
+		logger:              config.GetLoggerFor("Prometheus Service"),
+		gauges:              map[string]prometheus.Gauge{},
+		counters:            map[string]prometheus.Counter{},
+		isMonitoringEnabled: isMonitoringEnabled,
+		assetsConfig:        assetsConfig,
 	}
 }
 
-func (s Service) CreateAndRegisterGaugeMetric(name string, help string) prometheus.Gauge {
+func (s *Service) CreateAndRegisterGaugeMetric(name string, help string, labels prometheus.Labels) prometheus.Gauge {
```

Comment:
Can we rename `CreateAndRegisterGaugeMetric` to `CreateAndRegisterIfNotExistsGaugeMetric` and document it?

Reply:
okay, but at least to be `CreateAndRegisterGaugeMetricIfNotExists` as it is more correct.

Reply:
Having the `GaugeMetric` makes it more unreadable to figure out what metric you are creating